graph/src/data/subgraph/mod.rs

```diff
@@ -898,27 +900,51 @@ impl UnresolvedSubgraphManifest {
             repository,
             schema,
             data_sources,
+            templates,
         } = self;
 
+        match semver::Version::parse(&spec_version) {
+            Ok(ref ver) if *ver <= semver::Version::new(0, 0, 2) => {}
+            _ => {
+                return Box::new(future::err(format_err!(
+                    "This Graph Node only supports manifest spec versions <= 0.0.2,
+                but subgraph `{}` uses `{}`",
+                    id,
+                    spec_version
+                ))) as Box<dyn Future<Item = _, Error = _> + Send>;
+            }
+        }
+
         // resolve each data set
```

Comment:
I'm just noticing now that we're using `data set` and `data_set` in this method. Fancy changing that to `data source` / `data_source` (optional for this review)?